Added stylesheet

extremelypenguin opened the pull request:
Now that we've successfully added and called the stylesheet, we can get to the fun part - the CSS!

niccokunzmann commented on schul_cloud_resources_server_tests/app.py:
```diff
@@ -247,10 +252,14 @@ def delete_resources():
 
 @get("/")
 @get("/v1")
+@route("stylesheet.css")
```
I think, they should start with "/".
You can try to reach the style sheet by adding "/stylesheet.css" to the url and see what you get.

niccokunzmann commented:
If you like to have a look, I took another approach here: https://github.com/niccokunzmann/printathpi/blob/master/printathpi/app.py#L20 There is a static folder next to this file and this function served the content of /static.

extremelypenguin commented:
@niccokunzmann hmm, you're right, we should consider making a /static/ folder. the bottle.py docs also use a static folder, why is that?

niccokunzmann commented:
I think because it is convenient: When there is a static folder, one does not need to change the app to serve new files. They need to be copied to that folder and that is it.

extremelypenguin commented:
hmm, I've just added a commit but I'm not sure why its listing so many changed files

niccokunzmann commented:
I think, you copied the folder into itself :  schul_cloud_resources_server_tests/schul_cloud_resources_server_tests/.travis.yml should be .travis.yml

niccokunzmann commented:
Also, there are past changes in it. e.g. in the README.rst file - these are some I created in the past and they are there again. I am not sure why.

If you like to try out some git, you can
- checkout the master branch of this repository
- use `git cherry-pick <commit>` to get the commits you really want

extremelypenguin commented:
i don't need to cherrypick, the problem is that for some reason the called stylesheet commit has all those changed files. I've reverted the "attempted to call stylesheet but got error" commit because we don't need that, but we need to remove all the files from the "called stylesheet" commit. Cherrypicking is for entire commits, correct?

niccokunzmann commented:
That is correct. I think, your way is less work than what I suggested.

extremelypenguin commented:
you were right, I accidentally copied the folders into themselves. I have removed the files, so everything should be okay now

extremelypenguin commented:
it seems that the changes i've made to app.py aren't showing up

extremelypenguin commented:
just added a commit fixing everything, I believe everything should work now, it looks correct to me.

niccokunzmann commented on schul_cloud_resources_server_tests/app.py:
```diff
@@ -205,6 +206,11 @@ def add_resource():
+# call css stylesheet
+@route('/schul_cloud_resources_server_tests/<filepath>')
+def server_static(filepath):
+    return static_file(filepath, root="")
```
Setting root to "" can be a security risk: It might mean, you can read from anywhere. I do not know. Would it be ok for you to use `root=HERE`?
See the variable HERE which is the folder of this file: 
https://github.com/extremelypenguin/schul_cloud_resources_server_tests/blob/ab8aed6c5512c8934b5b3c1cbc345c3978691aaf/schul_cloud_resources_server_tests/app.py#L8

niccokunzmann commented:
@extremelypenguin I am excited: This looks like it works. I left a comment. What do you think, would you like to change something more or is this ready for merging?

extremelypenguin commented:
changed the root thing and tested it. I also added some stuff to the stylesheet to test if the stylesheet works, and everything works fine on my comptuter.

niccokunzmann commented:
@extremelypenguin Thank you very much! This is a foundation to add more design to the page!

niccokunzmann commented:
I should be deployed tomorrow.

extremelypenguin commented:
@niccokunzmann i just remembered, I made a commit to the stylesheet. The body is set to black and h1 is set to blue - this looks horrible and wasn't supposed to be deployed, it was just a test to see if the stylesheet was actually being called. You should probably fix that.
